File: threedollar-boss-domain/src/main/java/com/depromeet/threedollar/foodtruck/domain/domain/account/BossAccount.java
package com.depromeet.threedollar.foodtruck.domain.domain.account;

import com.depromeet.threedollar.foodtruck.domain.domain.common.BaseTimeEntity;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class BossAccount extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Embedded
    private BossSocialInfo socialInfo;

    private String name;

    private BossAccount(String socialId, BossAccountSocialType socialType, String name) {
        this.socialInfo = BossSocialInfo.of(socialId, socialType);
        this.name = name;
    }

    public static BossAccount of(String socialId, BossAccountSocialType socialType, String name) {
        return new BossAccount(socialId, socialType, name);
    }

}

File: threedollar-domain-mongo/src/main/kotlin/com/depromeet/threedollar/domain/mongo/boss/domain/account/BossAccount.kt
package com.depromeet.threedollar.domain.mongo.boss.domain.account

import com.depromeet.threedollar.domain.mongo.common.domain.BaseDocument
import com.depromeet.threedollar.domain.mongo.common.domain.BusinessNumber
import org.springframework.data.mongodb.core.mapping.Document

@Document("boss_account_v1")
class BossAccount(
    var name: String,
    val socialInfo: BossAccountSocialInfo,
    val businessNumber: BusinessNumber,
    var pushSettingsStatus: PushSettingsStatus
) : BaseDocument() {

    fun update(name: String, pushSettingsStatus: PushSettingsStatus) {
        this.name = name
        this.pushSettingsStatus = pushSettingsStatus
    }

    companion object {
        fun of(
            name: String,
            socialId: String,
            socialType: BossAccountSocialType,
            businessNumber: BusinessNumber,
            pushSettingsStatus: PushSettingsStatus = PushSettingsStatus.OFF
        ): BossAccount {
            return BossAccount(
                name = name,
                socialInfo = BossAccountSocialInfo.of(socialId, socialType),
                businessNumber = businessNumber,
                pushSettingsStatus = pushSettingsStatus
            )
        }
    }

}
